datasets/tedlium2.py

```diff
@@ -214,3 +214,115 @@ def load_stm_data(self, stm_file):
         return data
+
+
+class CreateTEDLIUM2BlissCorpusJobV2(CreateTEDLIUM2BlissCorpusJob):
+    """
+    It has the same logic as CreateTEDLIUM2BlissCorpusJob but the corpus name is distinct for each partition
+    and the common text normalization for eliminating space before apostroph, similarly to Kaldi and ESPNet is applied
+    """
+
+    def make_stm(self):
+        def extend_segment_time(seg, prev_seg, next_seg, start_pad=0.15, end_pad=0.1):
+            start = float(seg[3])
+            end = float(seg[4])
+            # start padding (previous seg alread padded)
+            if prev_seg is not None and seg[0] == prev_seg[0]:
+                prev_end = float(prev_seg[4])
+            else:
+                prev_end = 0.0
+            start = max(start - start_pad, prev_end)
+            # end padding (next seg not yet padded and start padding is more important)
+            next_start = end + end_pad
+            if next_seg is not None and seg[0] == next_seg[0]:
+                next_start = max(float(next_seg[3]) - start_pad, end)
+            end = min(end + end_pad, next_start)
+            return "%.2f" % (start), "%.2f" % (end)
+
+        header = [
+            ";;",
+            ';; LABEL "o" "Overall" "Overall results"',
+            ';; LABEL "f0" "f0" "Wideband channel"',
+            ';; LABEL "f2" "f2" "Telephone channel"',
+            ';; LABEL "male" "Male" "Male Talkers"',
+            ';; LABEL "female" "Female" "Female Talkers"',
+            ";;",
+        ]
+        for corpus_key in ["train", "dev", "test"]:
+            f = open("%s.stm" % corpus_key, "w")
+            f.write("\n".join(header) + "\n")
+
+            stm_folder = os.path.join(self.corpus_folders[corpus_key].get_path(), "stm")
+            for stm_file in sorted(os.listdir(stm_folder)):
+                if not stm_file.endswith(".stm"):
+                    continue
+                data = self.load_stm_data(os.path.join(stm_folder, stm_file))
+                for idx in range(len(data)):
+                    # some text normalization
+                    text = data[idx][6]
+                    text = text.replace("imiss", "i miss")
+                    text = text.replace("uptheir", "up their")
+                    data[idx][6] = text
+
+                    # train-only: segment boundary non-overlapping extension (kaldi)
+                    if corpus_key == "train":
+                        pre_seg = None if idx == 0 else data[idx - 1]
+                        next_seg = None if idx == len(data) - 1 else data[idx + 1]
+                        data[idx][3], data[idx][4] = extend_segment_time(data[idx], pre_seg, next_seg, 0.15, 0.1)
+
+                for seg in data:
+                    f.write(" ".join(seg) + "\n")
+
+            f.close()
+            shutil.move("%s.stm" % corpus_key, self.out_stm_files[corpus_key].get_path())
+
+    def make_corpus(self):
```

Comment:
Do you make any changes in this function? If not it can be removed.

Reply:
Yes, this function now does the name distinction for each partition, the original reason why this PR was opened :-)